### ISSUE TITLE
nginz upstreams sometimes get a wrong config

### ISSUE THREAD

**Issue by tiago-loureiro** [wireapp/wire-server-deploy#110]
Under certain circumstances, I have witnessed this on our logs:

Log from nginz:
```
Config file update detected
2019/07/09 06:44:50 [emerg] 12745#0: invalid number of arguments in "server" directive in /etc/wire/nginz/upstreams/upstreams.conf:30
nginx: [emerg] invalid number of arguments in "server" directive in /etc/wire/nginz/upstreams/upstreams.conf:30
nginx: configuration file /etc/wire/nginz/conf/nginx.conf test failed
ERROR: New configuration is invalid!!
Config file update detected
nginx: the configuration file /etc/wire/nginz/conf/nginx.conf syntax is ok
nginx: configuration file /etc/wire/nginz/conf/nginx.conf test is successful
New configuration is valid, reloading nginx
2019/07/09 06:44:52 [notice] 12747#0: signal process started
Config file update detected
2019/07/09 06:45:15 [emerg] 12756#0: invalid number of arguments in "server" directive in /etc/wire/nginz/upstreams/upstreams.conf:40
nginx: [emerg] invalid number of arguments in "server" directive in /etc/wire/nginz/upstreams/upstreams.conf:40
nginx: configuration file /etc/wire/nginz/conf/nginx.conf test failed
ERROR: New configuration is invalid!!
Config file update detected
nginx: the configuration file /etc/wire/nginz/conf/nginx.conf syntax is ok
nginx: configuration file /etc/wire/nginz/conf/nginx.conf test is successful
New configuration is valid, reloading nginx
2019/07/09 06:45:17 [notice] 12758#0: signal process started
```

This is how one of these problematic configs looks like:
```
upstream brig {
         least_conn;
         keepalive 32;
         server 10.233.X.X:8080 max_fails=3 weight=100;
}
upstream cannon {
         least_conn;
         keepalive 32;
         server 10.233.X.X:8080 max_fails=3 weight=100;
}
upstream cargohold {
         least_conn;
         keepalive 32;
         server 10.233.X.X:8080 max_fails=3 weight=100;
}
upstream galley {
         least_conn;
         keepalive 32;
         server 10.233.X.X:8080 max_fails=3 weight=100;
}
upstream gundeck {
         least_conn;
         keepalive 32;
         server 10.233.X.X:8080 max_fails=3 weight=100;
}
upstream ibis {
         least_conn;
         keepalive 32;
         server localhost:8080 down;
}
upstream proxy {
         least_conn;
         keepalive 32;
         server ;;:8080 max_fails=3 weight=100;
         server expected:8080 max_fails=3 weight=100;
         server opt:8080 max_fails=3 weight=100;
         server record:8080 max_fails=3 weight=100;
         server in:8080 max_fails=3 weight=100;
         server response:8080 max_fails=3 weight=100;
}
upstream spar {
         least_conn;
         keepalive 32;
         server localhost:8080 down;
```

Not really sure how we end up in such state but worth putting this here as it has been seen multiple times.

**Comment by jschaul** [comment]
Interesting. This looks like the nginz-disco (https://github.com/wireapp/wire-server/pull/759/files#diff-8df8c154b88bbc3fd7bcdb00f67cf403R16) is not stopping when dig fails to find IPs and instead inserts garbage. Options:

1. teach nginz-disco/bash what an IP should look like, and stop when it's not an IP
2. rewrite nginz-disco in e.g. Haskell and get better types support.

**Comment by jschaul** [comment]
check last commit on https://github.com/wireapp/wire-server/pull/759

**Comment by jschaul** [comment]
This should be fixed with #125 (that contains a few sanity checks).
